addons/product/views/product_views.xml

```diff
@@ -173,7 +173,7 @@
                 <separator/>
                 <filter string="Archived" name="inactive" domain="[('active','=',False)]"/>
                 <group expand="1" string="Group By">
-                    <filter string="Product Type" name="type" context="{'group_by':'type'}"/>
+                    <filter string="Product Type" name="detailed_type" context="{'group_by':'detailed_type'}"/>
```

Comment:
The name shouldn't be changed in a stable version as inheriting views may have used it as a hook.